katdal/test/test_datasources.py

```diff
@@ -187,3 +244,34 @@ def test_missing_chunks_uneven_chunking(self):
                 'weights_channel': (1, 7),
                 'flags': (4, 15, 30)
             })
+
+
+class TestTelstateDataSource(object):
+    def setup(self):
+        self.tempdir = tempfile.mkdtemp()
+        self.store = NpyFileChunkStore(self.tempdir)
```

Comment:
What about a `DictChunkStore` to avoid temp files? Although, NPY's are probably more representative...

Reply:
DictChunkStore is read-only, so wouldn't work with `put_fake_dataset`.